src/core/src/pass/serialize.cpp

```diff
@@ -132,14 +132,14 @@ class ConstantWriter {
             // Therefore we always have to compare values when finding a match in the hash multimap.
             const HashValue hash = ov::runtime::compute_hash(ptr_to_write, new_size);
 
-            auto found = m_hash_to_file_positions.find(hash);
+            auto found = m_hash_to_file_positions.equal_range(hash);
```

Comment:
is it possible to add a test?

Reply:
How about limiting number of attempts by some reasonable number? Are hashed and the current data guaranteed to have the same size?

Reply:
It turned out we already had tests for that. Added more clever comparison, blobs are compared only if sizes match as well.